mne/time_frequency/tests/test_psd.py

```diff
@@ -14,99 +16,102 @@
+
+    # -- Raw --
+    picks_psd = picks[:2]
+    kws_psd = dict(tmin=tmin, tmax=tmax, fmin=fmin,
+                   fmax=fmax, picks=picks_psd)  # Common to all
+    kws_welch = dict(n_fft=n_fft)
+    kws_mt = dict(low_bias=True)
+    funcs = {psd_welch: kws_welch,
+             psd_multitaper: kws_mt,
+             compute_raw_psd: kws_welch}
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        for func, kws in funcs.iteritems():
+            kws = kws.copy()
+            kws.update(kws_psd)
+            psds, freqs = func(raw, proj=False, **kws)
+            psds_proj, freqs_proj = func(raw, proj=True, **kws)
+
+            assert_array_almost_equal(psds, psds_proj)
+            assert_true(psds.shape == (len(kws['picks']), len(freqs)))
+            assert_true(np.sum(freqs < 0) == 0)
+            assert_true(np.sum(psds < 0) == 0)
+
+            # Array input shouldn't work
+            assert_raises(ValueError, func, raw[:3, :20][0])
+        assert_true(len(w), 3)
+
+    # -- Epochs/Evoked --
     events = read_events(event_fname)
-
+    tmin, tmax, event_id = -0.5, 0.5, 1
     epochs = Epochs(raw, events[:10], event_id, tmin, tmax, picks=picks,
                     baseline=(None, 0),
                     reject=dict(grad=4000e-13, eog=150e-6), proj=False,
                     preload=True)
+    evoked = epochs.average()
 
     tmin_full, tmax_full = -1, 1
-    epochs_full = Epochs(raw, events[:10], event_id, tmax=tmax_full,
-                         tmin=tmin_full, picks=picks,
-                         baseline=(None, 0),
+    epochs_full = Epochs(raw, events[:10], event_id, tmin_full, tmax_full,
+                         picks=picks, baseline=(None, 0),
                          reject=dict(grad=4000e-13, eog=150e-6), proj=False,
                          preload=True)
 
-    picks = pick_types(epochs.info, meg='grad', eeg=False, eog=True,
-                       stim=False, include=include, exclude='bads')
-    psds, freqs = compute_epochs_psd(epochs[:1], fmin=2, fmax=300,
-                                     n_fft=n_fft, picks=picks)
-
-    psds_t, freqs_t = compute_epochs_psd(epochs_full[:1], fmin=2, fmax=300,
-                                         tmin=tmin, tmax=tmax,
-                                         n_fft=n_fft, picks=picks)
-    # this one will fail if you add for example 0.1 to tmin
-    assert_array_almost_equal(psds, psds_t, 27)
-
-    psds_proj, _ = compute_epochs_psd(epochs[:1].apply_proj(), fmin=2,
-                                      fmax=300, n_fft=n_fft, picks=picks)
-
-    assert_array_almost_equal(psds, psds_proj)
-    assert_true(psds.shape == (1, len(picks), len(freqs)))
-    assert_true(np.sum(freqs < 0) == 0)
-    assert_true(np.sum(psds < 0) == 0)
+    kws_psd = dict(fmin=fmin, fmax=fmax, picks=picks_psd)  # Common to all
+    kws_welch = dict(n_fft=n_fft)
+    kws_mt = dict(low_bias=True)
```

Comment:
ah - you're right that kws_welch/mt are duplicated. I need to redefine kws_psd because now tmin/tmax have changed, that way I don't have to define them in each call to the function. That's not reflected in this code but I'll push it in a sec.